### PR TITLE
Docs, part 8

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #358

- `ObjectParser` - added docs.
- `ObjectParser` - made `getReflectionProperties()` method private.
- `ObjectParser` - renamed `reflection` cache item to `reflectionSource` to be more clear.
- `ObjectParser` - fixed error when cache key was the same regardless of `$skipStaticProperties` option.
- `ObjectParser` - improved typing for attribute targets.
- `ArrayDataSet` - fixed typo.
- Attribute translation - fixed docs (grammar).
- Moved data classes used in tests for PHP 8.1 to according folder.

### CODE REVIEW

File: src/Helper/ObjectParser.php [hunk 1]
```diff
@@ -27,22 +103,43 @@ final class ObjectParser
...
+     */
     private string|null $cacheKey;
 
     public function __construct(
+        /**
+         * @var object An object for parsing rules and data.
+         */
         private object $object,
+        /**
+         * @var int A constant determining what visibility levels the parsed properties must have. For example: public
```
Comment (samdark) [inline src/Helper/ObjectParser.php:117]:
```suggestion
         * @var int Visibility levels the parsed properties must have. For example: public
```

File: src/Helper/ObjectParser.php [hunk 2]
```diff
@@ -27,22 +103,43 @@ final class ObjectParser
...
+         * and protected only, this means that the rest (private ones) will be skipped. Defaults to all visibility
+         * levels (public, protected and private).
+         */
         private int $propertyVisibility = ReflectionProperty::IS_PRIVATE |
         ReflectionProperty::IS_PROTECTED |
         ReflectionProperty::IS_PUBLIC,
+        /**
+         * @var bool A flag indicating whether the properties with "static" modifier must be skipped.
```
Comment (samdark) [inline src/Helper/ObjectParser.php:125]:
```suggestion
         * @var bool Whether the properties with "static" modifier must be skipped.
```

File: src/Helper/ObjectParser.php [hunk 3]
```diff
@@ -27,22 +103,43 @@ final class ObjectParser
...
+         */
         private bool $skipStaticProperties = false,
-        bool $useCache = true
+        /**
+         * @var bool A flag indicating whether some results of parsing (Reflection of object with properties and
```
Comment (samdark) [inline src/Helper/ObjectParser.php:129]:
```suggestion
         * @var bool Whether some results of parsing (Reflection of object with properties and
```

File: src/Helper/ObjectParser.php [hunk 4]
```diff
@@ -72,23 +169,42 @@ public function getRules(): array
...
-        if ($this->useCache()) {
-            $this->setCacheItem('rules', $rules);
-        }
+        $this->setCacheItem('rules', $rules);
 
         return $rules;
     }
 
+    /**
+     * Returns a property value of the parsed object. Defaults are handled too.
```
Comment (samdark) [inline src/Helper/ObjectParser.php:178]:
```suggestion
     * Returns a property value of the parsed object.
```

File: src/Helper/ObjectParser.php [hunk 5]
```diff
@@ -72,23 +169,42 @@ public function getRules(): array
...
+     *
+     * Note that in case of non-existing property a default `null` value is returned. If you need to check the presence
+     * of property or return a different default value, use {@see hasAttribute()} instead.
```
Comment (samdark) [inline src/Helper/ObjectParser.php:181]:
```suggestion
     * of a property or return a different default value, use {@see hasAttribute()} instead.
```

**Comment by arogachev** [comment]
Accidentally clicked "merge" instead of "ready for review". Sorry. Reverted. A new PR - #458.

**Pull request merged**